src/hardware/joystick.cpp

```diff
@@ -404,6 +413,131 @@ void JOYSTICK_ParseConfiguredType()
 	assert(joytype != JOY_UNSET);
 }
 
+enum CalibrationType {
+	NegativeScale,
+	PositiveScale,
+	NegativeOffset,
+	PositiveOffset,
+};
+
+static void calibrate_axis(const bool pressed, AxisRateConstants &rates,
+                           const CalibrationType calibration_type)
+{
+	if (!pressed)
+		return;
+
+	constexpr auto negative_scale = 0.95;
+	constexpr auto positive_scale = 1.05;
+	constexpr auto delta_offset   = 0.005;
```

Comment:
I don't know the answer myself, just wondering if these scaling factors as fine-grained enough? 5% change, that can be pretty big when it comes to calibration, no? I guess we could get user feedback and make it finer if it's deemed to be too coarse?

Reply:
Yup, 5% is about right, from the testing I've done. Program is attached below, with a canned conf for you to try it (need a joystick plugged in, with the analog stick active instead of the d-pad).